endpointregistry/epregistry.go

```diff
@@ -0,0 +1,62 @@
+package endpointregistry
+
+import (
+	"sync"
+	"time"
+)
+
+type entry struct {
+	detected         time.Time
+	inflightRequests int64
+}
+
+var data map[string]entry = map[string]entry{}
+var m sync.Mutex
+
+func Clear() {
+	m.Lock()
+	defer m.Unlock()
+
+	data = map[string]entry{}
+}
+
+func SetDetectedTime(key string, detected time.Time) {
+	m.Lock()
+	defer m.Unlock()
+
+	e := data[key]
+	e.detected = detected
+	data[key] = e
+}
+
+func GetDetectedTime(key string) time.Time {
+	m.Lock()
+	defer m.Unlock()
+
+	return data[key].detected
+}
+
+func GetInflightRequests(key string) int {
```

Comment:
Why int?

Reply:
@AlexanderYastrebov do you have partial inflight requests that you would count with fractions of int?

Reply:
I mean why we cast int64 to int here